search/search_repository.go

```diff
@@ -615,11 +615,11 @@ func (r *GormSearchRepository) listItemsFromDB(ctx context.Context, criteria cri
 
 	if parentExists != nil && !*parentExists {
 		where += ` AND
-			id not in (
-				SELECT target_id FROM work_item_links
-				WHERE link_type_id IN (
-					SELECT id FROM work_item_link_types WHERE forward_name = 'parent of'
-				)
+			NOT EXISTS (
+				SELECT wil.target_id FROM work_item_links wil, work_item_link_types wilt
+				WHERE wil.link_type_id = wilt.id AND wilt.forward_name = 'parent of' 
+				AND wil.target_id = work_items.id
+				AND wil.deleted_at IS NULL
```

Comment:
Yes, the main change is the addition of `wil.deleted_at IS NULL` to avoid taking into account deleted links. 
Good point about the use of the constant, I'll fix that.
I personally think that the `NOT EXISTS` is better to read and is supposed to be more efficient, too.

Reply:
I'd say it is not more efficient because the subquery will only executed once and you perform a full join, I guess.